test/robustness/overlay/buffer/recursive_polygons_buffer.cpp

```diff
@@ -79,45 +79,54 @@ void create_svg(std::string const& filename
     mapper.map(buffer, "stroke-opacity:0.9;stroke:rgb(0,0,0);fill:none;stroke-width:1");
 }
 
-template <typename MultiPolygon, typename Settings>
-bool verify(std::string const& caseid, MultiPolygon const& mp, MultiPolygon const& buffer, Settings const& settings)
+template <typename Geometry, typename Buffer>
+bool verify_buffer(Geometry const& geometry, Buffer const& buffer, std::string& reason, bool check_validity)
```

Comment:
This is a stricter check in the test program, and gives more structured info.
It still happens a lot (around `0.25%` of the cases is wrong)